backend/dataall/modules/dataset_sharing/services/share_object_service.py

```diff
@@ -42,12 +44,33 @@
 from dataall.modules.datasets_base.db.dataset_repositories import DatasetRepository
 from dataall.modules.datasets_base.db.dataset_models import DatasetTable, Dataset
 from dataall.modules.datasets_base.services.permissions import DATASET_TABLE_READ
+from dataall.base.aws.iam import IAM
+from dataall.base.aws.sts import SessionHelper
+
 import logging
 
 log = logging.getLogger(__name__)
 
 
 class ShareObjectService:
+    @staticmethod
+    def verify_principal_role(session, share: ShareObject) -> bool:
+        role_name = share.principalIAMRoleName
+        account_id = SessionHelper.get_account()
+        delegation_role = SessionHelper.get_delegation_role_arn(account_id)
+        principal_role = IAM.get_role_arn_by_name(account_id, role_name, delegation_role)
+        if not principal_role:
+            for item in ShareObjectRepository.get_all_sharable_items(session, share.shareUri):
+                ShareObjectRepository.update_share_item_health_status(
+                    session,
+                    share_item=item,
+                    healthStatus=ShareItemHealthStatus.Unhealthy.value,
+                    healthMessage=f'Share principal role {role_name} not found.',
+                    timestamp=datetime.now(),
+                )
+            return False
```

Comment:
This function is doing 2 things: it is verifying the principal IAM role and then it is updating the health status, this second part might be problematic. Health-statuses are only changed in the verification workflows of the item. If we start adding changes in other parts of the code, it will soon start be difficult to follow (very similar to what happened with the regular share workflows in the past).

Reply:
With that in mind, I would break it down and only do the verification of the "does IAM role exist?" here. Which leads to the idea of, do we even need to make it a service function of sharing? We already have `base/aws.iam.IAM.get_role_by_name` that we can call directly, and as explained before it is just a:

```
environment = EnvironmentService.get_environment_by_uri(session, share.environmentUri)
principal_role = IAM.get_role_arn_by_name(
                account=environment.AwsAccountId,
                region=environment.region,
                role_name=share.principalIAMRoleName
                )
```

Reply:
adressed in [6e5951e](https://github.com/data-dot-all/dataall/pull/1161/commits/6e5951e7a849cdb0230f27379d3523fe2ace96b8)